File: API/APIRest2_23_24/src/main/java/es/luisherrero/apirest1/model/Incidencia.java
package es.luisherrero.apirest1.model;

import java.io.Serializable;
import java.sql.Time;
import java.util.Date;

import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@jakarta.persistence.Entity
@Table(name = "incidencias")
public class Incidencia implements Serializable {
	private static final long serialVersionUID = 1;

	@Id
	private int num;

	private String adjuntoUrl;

	private String descripcion;

	@Enumerated(EnumType.STRING)
	private Estado_Incidencia estado;

	private Date fechaCierre;

	private Date fechaCreacion;

	private Time tiempo_dec;
	
	@Enumerated(EnumType.STRING)
	private Tipo_Incidencias tipo;

	// private List<Comentario> comentarios;

	@ManyToOne
	//@Column(name = "equipo_id")
	private Equipo equipo;

	@ManyToOne
	@JoinColumn(name="subtipo_id")
	private IncidenciasSubtipo incidenciasSubtipo;

	@ManyToOne
	private Personal creador;

	@ManyToOne
	private Personal responsable;

	public Incidencia() {
	}

	public int getNum() {
		return this.num;
	}

	public void setNum(int num) {
		this.num = num;
	}

	public String getAdjuntoUrl() {
		return this.adjuntoUrl;
	}

	public void setAdjuntoUrl(String adjuntoUrl) {
		this.adjuntoUrl = adjuntoUrl;
	}

	public String getDescripcion() {
		return this.descripcion;
	}

	public void setDescripcion(String descripcion) {
		this.descripcion = descripcion;
	}

	public Estado_Incidencia getEstado() {
		return this.estado;
	}

	public void setEstado(Estado_Incidencia estado) {
		this.estado = estado;
	}

	public Date getFechaCierre() {
		return this.fechaCierre;
	}

	public void setFechaCierre(Date fechaCierre) {
		this.fechaCierre = fechaCierre;
	}

	public Date getFechaCreacion() {
		return this.fechaCreacion;
	}

	public void setFechaCreacion(Date fechaCreacion) {
		this.fechaCreacion = fechaCreacion;
	}

	public Tipo_Incidencias getTipo() {
		return this.tipo;
	}

	public void setTipo(Tipo_Incidencias tipo) {
		this.tipo = tipo;
	}

	/*
	 * public List<Comentario> getComentarios() { return this.comentarios; }
	 * 
	 * public void setComentarios(List<Comentario> comentarios) { this.comentarios =
	 * comentarios; }
	 * 
	 * public Comentario addComentario(Comentario comentario) {
	 * getComentarios().add(comentario); comentario.setIncidencia(this);
	 * 
	 * return comentario; }
	 * 
	 * public Comentario removeComentario(Comentario comentario) {
	 * getComentarios().remove(comentario); comentario.setIncidencia(null);
	 * 
	 * return comentario;
	 */
	public Equipo getEquipo() {
		return this.equipo;
	}

	public void setEquipo(Equipo equipo) {
		this.equipo = equipo;
	}

	public IncidenciasSubtipo getIncidenciasSubtipo() {
		return this.incidenciasSubtipo;
	}

	public void setIncidenciasSubtipo(IncidenciasSubtipo incidenciasSubtipo) {
		this.incidenciasSubtipo = incidenciasSubtipo;
	}

	public Personal getCreador() {
		return this.creador;
	}

	public void setCreador(Personal creador) {
		this.creador = creador;
	}

	public Personal getResponsable() {
		return this.responsable;
	}

	public void setResponsable(Personal creador) {
		this.responsable = creador;
	}

	public Time getTiempo_dec() {
		return tiempo_dec;
	}

	public void setTiempo_dec(Time tiempo_dec) {
		this.tiempo_dec = tiempo_dec;
	}
	
	
}

File: Creacion API/Python_e_imagenes/incidencia.py
class Incidencia:
    def __init__(self, num, tipo, subtipo_id, fecha_creacion, fecha_cierre, descripcion, estado, adjunto_url, creador_id, responsable_id, equipo_id):
        self.num = num
        self.tipo = tipo
        self.subtipo_id = subtipo_id
        self.fecha_creacion = fecha_creacion
        self.fecha_cierre = fecha_cierre
        self.descripcion = descripcion
        self.estado = estado
        self.adjunto_url = adjunto_url
        self.creador_id = creador_id
        self.responsable_id = responsable_id
        self.equipo_id = equipo_id

    def get_num(self):
        return self.num

    def set_num(self, value):
        self.num = value
        
    def get_tipo(self):
        return self.tipo

    def set_tipo(self, value):
        self.tipo = value
    
    def get_subtipo_id(self):
        return self.subtipo_id

    def set_subtipo_id(self, value):
        self.subtipo_id = value
    
    def get_fecha_creacion(self):
        return self.fecha_creacion

    def set_fecha_creacion(self, value):
        self.fecha_creacion = value
    
    def get_fecha_cierre(self):
        return self.fecha_cierre

    def set_fecha_cierre(self, value):
        self.fecha_cierre = value
        
    def get_descripcion(self):
        return self.descripcion

    def set_descripcion(self, value):
        self.descripcion = value
        
    def get_estado(self):
        return self.estado

    def set_estado(self, value):
        self.estado = value
        
    def get_adjunto_url(self):
        return self.adjunto_url

    def set_adjunto_url(self, value):
        self.adjunto_url = value
        
    def get_creador_id(self):
        return self.creador_id

    def set_creador_id(self, value):
        self.creador_id = value
        
    def get_responsable_id(self):
        return self.responsable_id

    def set_responsable_id(self, value):
        self.responsable_id = value
        
    def get_equipo_id(self):
        return self.equipo_id

    def set_equipo_id(self, value):
        self.equipo_id = value